### ISSUE TITLE
Seems like the batch labels and bbox are using the wrong indexes

### ISSUE THREAD

**Issue by manojpatel** [dongjk/faster_rcnn_keras#17]
`batch_inds=inds_inside[labels!=-1]`
`batch_inds=(batch_inds / k).astype(np.int)`
these are all tiles for which few boxes get max overlap
`batch_label_targets=full_labels.reshape(-1,1,1,1*k)[batch_inds]`
it should be
`batch_label_targets=full_labels.reshape(-1,1,1,1*k)[inds_inside[labels!=-1]]`
The same way
`batch_bbox_targets = bbox_targets.reshape(-1,1,1,4*k)[batch_inds]`
it should be
`batch_bbox_targets = bbox_targets.reshape(-1,1,1,4*k)[inds_inside[labels!=-1]]`

`batch_inds`  are tile index but we needs to pick the anchor index.
Please let me know if I am missing any thing here.

**Comment by ehijano** [comment]
No, It has to be the feature point index. Note that in your arrays, you are reshaping to (-1,1,1,1*k) and (-1,1,1,4*k), so the first index in each list refers to a point in each feature map, containing 1*k scores and 4*k anchor targets, corresponding to the score and anchor target of each anchor in the feature map point.

**Comment by manojpatel** [comment]
Its my bad, would say silly, thanks for pointing. I am not finding the data at the given location https://www.kaggle.com/c/imagenet-object-detection-challenge/data. Can you please help to know if you have a link for the same.

One more quick question for
```
def clip_boxes(boxes, im_shape):
    # x1 >= 0
    boxes[:, 0::4] = np.maximum(np.minimum(boxes[:, 0::4], im_shape[1] - 1), 0)
    # y1 >= 0
    boxes[:, 1::4] = np.maximum(np.minimum(boxes[:, 1::4], im_shape[0] - 1), 0)
    # x2 < im_shape[1]
    boxes[:, 2::4] = np.maximum(np.minimum(boxes[:, 2::4], im_shape[1] - 1), 0)
    # y2 < im_shape[0]
    boxes[:, 3::4] = np.maximum(np.minimum(boxes[:, 3::4], im_shape[0] - 1), 0)
    return boxes
```
My understanding is im_shape is height and width scaling factor.
Does the function trying to clip Xmax and Ymax?
